### PR TITLE
fix(secu): protect from SQL injections hostXML.php for 18 10

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

An authenticated attacker can perform an SQL injection on this page parameters

**Fixes** # (CVE-2019-??)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [x] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please contact me.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 1]
```diff
@@ -70,187 +67,221 @@
...
+$instance = filter_var($_GET['instance'] ?? $obj->defaultPoller, FILTER_VALIDATE_INT);
+$hostgroup = filter_var($_GET['hostgroups'] ?? $obj->defaultHostgroups, FILTER_VALIDATE_INT);
+$search = filter_var($_GET['search'] ?? "", FILTER_SANITIZE_STRING);
+$order = isset($_GET['order']) && $_GET['order'] === "ASC"
+    ? "ASC"
+    : "DESC";
+$dateFormat = filter_var($_GET['date_time_format_status'] ?? "Y/m/d H:i:s", FILTER_SANITIZE_STRING);
+$statusHost = filter_var($_GET['statusHost'] ?? "", FILTER_SANITIZE_STRING);
+$statusFilter = filter_var($_GET['statusFilter'] ?? "", FILTER_SANITIZE_STRING);
+$criticalityValue = filter_var($_GET['criticality'] ?? $obj->defaultCriticality, FILTER_SANITIZE_STRING);
```
Comment (callapa) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]:
```suggestion
$criticalityValue = filter_var($_GET['criticality'] ?? $obj->defaultCriticality, FILTER_VALIDATE_INT);
```

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 2]
```diff
@@ -70,187 +67,221 @@
...
-
+//chosen criticality
+if ($criticalityValue) {
+    $stmt .= "customvariables cvs, ";
+    $whereClauses .= " AND h.host_id = cvs.host_id
+        AND cvs.name = 'CRITICALITY_ID'
+        AND cvs.service_id IS NULL
+        AND cvs.value = :criticalityValue";
+    $queryValues['criticalityValue'] = [
+        \PDO::PARAM_STR => CentreonUtils::escapeSecure($criticalityValue)
```
Comment (callapa) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]:
```suggestion
        \PDO::PARAM_INT => $criticalityValue
```

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 3]
```diff
@@ -70,187 +67,221 @@
...
+    $whereClauses .= " AND h.host_id = hhg.host_id
+        AND hg.hostgroup_id = :hostgroup
+        AND hhg.hostgroup_id = hg.hostgroup_id";
+    $queryValues['hostgroup'] = [
+        \PDO::PARAM_INT => $hostgroup
+    ];
 }
-
+//chosen criticality
+if ($criticalityValue) {
```
Comment (callapa) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]:
```suggestion
if ($criticalityValue !== false) {
```

File: www/include/monitoring/status/Hosts/xml/hostXML.php (hunk 1 above)
Comment (sc979) [reply www/include/monitoring/status/Hosts/xml/hostXML.php]: please check this line : https://github.com/centreon/centreon/pull/8006/files#diff-1dcae0c7db6caca11468504b47d83e29R179
this variable is a string ->  `AND cvs.value = :criticalityValue";`
as you can check it in the DB

Comment (sc979) [reply www/include/monitoring/status/Hosts/xml/hostXML.php]: please check this line : https://github.com/centreon/centreon/pull/8006/files#diff-1dcae0c7db6caca11468504b47d83e29R179
this variable is a string ->  `AND cvs.value = :criticalityValue";`
as you can check it in the DB

File: www/include/monitoring/status/Hosts/xml/hostXML.php (hunk 2 above)
Comment (sc979) [reply www/include/monitoring/status/Hosts/xml/hostXML.php]: same answer

**Pull request merged**